docs/source/cpp_api/core_resources.rst

```diff
@@ -55,6 +55,20 @@ namespace *raft::core*
     :project: RAFT
     :members:
 
+SNMG Device Resources
+---------------------
+
+The `raft::device_resources_snmg` provides a convenient way
+to setup a NCCL clique and access device-related resources on each device.
```

Comment:
Maybe a little more description here about what this means. What does SNMG mean? How do we expect it to be used? What are the imolications/caveats? I think it's safe to mention NCCL here, since after all these years we don't really have an alternative.